Add log level and custom option pipe mount arguments

tcibinan opened the pull request:
Relates to #1030.

The pull request improves pipe fuse integration in several ways in order to simplify troubleshooting.

## Logging level

The pull request brings support for an additional `pipe mount` argument `--log-level` which can be used to specify python [logging level](https://docs.python.org/2.7/howto/logging.html) which defaults to `ERROR`.

```bash
pipe storage mount --file \
                   --log-file log.txt \
                   --log-level INFO \
                   mountpoint
```

For the simplest cases `INFO` logging level should be enough. It registers all the file system events but `write` and `read` calls which are available only in `DEBUG` logging level mode. The `DEBUG` logging level collects all the available logs which can results in very large logs file especially if `threading` is enabled. Moreover pipe fuse performance is significantly worse in case of `DEBUG` logging level.

## Libfuse logging

There is a possibility to enable libfuse low level logging with fuse `debug` option. The pull request doesn't introduce the feature but it seems fine to point it out along other logging configurations.

```bash
pipe storage mount --file \
                   --options debug \
                   mountpoint
```

## Custom options

Moreover the pull request brings an additional `--custom-options` argument which can be used to fine tune pipe fuse behavior. All the custom options are transferred to pipe fuse command line arguments.

```bash
pipe storage mount --file \
                   --custom-options buffer-size=10000,cache-size=10 \
                   mountpoint
```

tcibinan committed:
Add log level argument to pipe mount

tcibinan committed:
Add custom options argument to pipe mount

tcibinan committed:
Add recording file system wrapper

tcibinan committed:
Repair frozen mounting

tcibinan committed:
Trim actual bytes from file system recording

tcibinan committed:
Set botocore logging level explicitly

tcibinan committed:
Remove too frequent logging call

tcibinan committed:
Log bytes length during file system recording

tcibinan committed:
Add more info logging

tcibinan committed:
Improve file system recording

tcibinan committed:
Record getxattr calls only in debug mode

mzueva reviewed (approved):
I'd suggest to refactor `mount` methods in pipe cli, but in a separate branch

NShaforostov commented:
I've tried to verify. Works well.

Minor note: might be, it would be useful to add possible values of log-levels into the help of the `storage mount` command?

tcibinan committed:
Improve pipe storage mount command help

tcibinan commented:
@NShaforostov Thanks for checking it out. Added available logging levels to `pipe storage mount` command's help message. Also cleaned up the help message a bit.
